packages/browser-wallet/src/popup/shared/account-info-emitter.ts

```diff
@@ -31,43 +24,47 @@ export class AccountInfoEmitter extends EventEmitter {
+        accountAddress: string,
+        listener: (accountInfo: AccountInfo) => void
+    ): (accountInfo: AccountInfo) => void {
+        if (!this.accounts.some((a) => a.address === accountAddress)) {
+            const addressObj = new AccountAddress(accountAddress);
+            this.accounts.push(addressObj);
         }
+        this.on(accountAddress, listener);
+        return listener;
+    }
 
-        // Note that the calls to get the account info is intentionally done
-        // in serial here (and in the interval) to limit the load on the server.
-        // If optimizing this to happen in parallel, then it must be ensured that
-        // the server can handle the number of requests received at the same time.
-
-        try {
-            const lastFinalizedBlockHash = (await this.client.getConsensusStatus()).lastFinalizedBlock;
-            for (const accountAddress of accountAddresses) {
-                const accountAddressObject = new AccountAddress(accountAddress);
-                const accountInfo = await this.client.getAccountInfo(accountAddressObject, lastFinalizedBlockHash);
-                if (accountInfo) {
-                    this.emit('accountinfo', accountInfo, accountAddress);
-                    this.emit('totalchanged', accountInfo, accountAddress);
-                }
-            }
-        } catch (e) {
-            this.emit('error', e);
+    unsubscribe(
+        accountAddress: string,
+        listener: (accountInfo: AccountInfo) => void
+    ): (accountInfo: AccountInfo) => void {
+        this.removeListener(accountAddress, listener);
+        if (this.listenerCount(accountAddress) === 0) {
+            this.accounts = this.accounts.filter((address) => accountAddress !== address.address);
         }
+        return listener;
+    }
 
+    async listen() {
         this.interval = setInterval(async () => {
             try {
-                const lfBlockHash = (await this.client.getConsensusStatus()).lastFinalizedBlock;
-                for (const address of this.accounts) {
-                    const accountInfo = await this.client.getAccountInfo(address, lfBlockHash);
-                    if (accountInfo) {
-                        this.emit('accountinfo', accountInfo, address.address);
-                        if (accountInfo.accountAmount !== this.previousTotalMap.get(address.address)) {
-                            this.emit('totalchanged', accountInfo, address.address);
-                            this.previousTotalMap.set(address.address, accountInfo.accountAmount);
+                if (this.accounts.length > 0) {
+                    const lfBlockHash = (await this.client.getConsensusStatus()).lastFinalizedBlock;
+                    for (const address of this.accounts) {
+                        const accountInfo = await this.client.getAccountInfo(address, lfBlockHash);
+                        if (accountInfo) {
+                            const network = await storedCurrentNetwork.get();
```

Comment:
Maybe we can avoid reading this from memory for each account?

Reply:
Actually this value was unused as we use `getGenesisHash` for the indexed storage. I have, however, moved that out of the loop to avoid this issue.